unity/PingPongGameExample/PingPongGameSkeleton/Assets/Scripts/Integration/PlatformIntegration/NetInterfaceIntegration.cs

```diff
@@ -0,0 +1,178 @@
+      return networkInterface;
+    }
+
+    IntPtr GetInetAddress(AndroidJavaObject networkInterface)
+    {
+      AndroidJavaObject inetAddresses = networkInterface.Call<AndroidJavaObject>("getInetAddresses");
+      IntPtr inetAddressesRaw = inetAddresses.GetRawObject();       // Get pointer to inetAddresses (Java enum), so that we can iterate through it
+      IntPtr inetAddressesClass = AndroidJNI.GetObjectClass(inetAddressesRaw);
+      // Null pointer??
+      IntPtr nextMethod = AndroidJNIHelper.GetMethodID(inetAddressesClass, "nextElement");
+
+      IntPtr inetAddress = AndroidJNI.CallObjectMethod(inetAddressesRaw, nextMethod, new jvalue[] { });
+      return inetAddress;
+    }
+
+    string GetHostAddress(IntPtr inetAddress)
+    {
+      IntPtr inetAddressClass = AndroidJNI.GetObjectClass(inetAddress);
+      IntPtr getHostAddressMethod = AndroidJNIHelper.GetMethodID(inetAddressClass, "getHostAddress");
+
+      String ipAddress = AndroidJNI.CallStringMethod(inetAddress, getHostAddressMethod, new jvalue[] { });
+      return ipAddress;
+    }
+
+    string GetNetworkInterfaceIP(String netInterfaceType)
+    {
+      AndroidJavaObject networkInterface = GetNetworkInterface(netInterfaceType);
+      if (networkInterface == null)
+      {
+        return "";
+      }
+      IntPtr inetAddress = GetInetAddress(networkInterface);
+      String ipAddress = GetHostAddress(inetAddress);
+      return ipAddress;
+    }
+
+    public string GetIPAddress(String netInterfaceType, AddressFamily adressFamily = AddressFamily.InterNetwork)
+    {
+      return GetNetworkInterfaceIP(netInterfaceType);
+    }
+
+    public bool HasWifi()
+    {
+      if (GetNetworkInterfaceIP(name.WIFI) != "")
+      {
+        return true;
+      }
+      return false;
+    }
+
+    public bool HasCellular()
+    {
+      if (GetNetworkInterfaceIP(name.CELLULAR) != "")
+      {
+        return true;
+      }
+      return false;
+    }
+
+#elif UNITY_IOS
+
+    [DllImport("__Internal")]
+    private static extern string _getIPAddress(string netInterfaceType);
+
+    [DllImport("__Internal")]
+    private static extern bool _isWifi();
+
+    [DllImport("__Internal")]
+    private static extern bool _isCellular();
+
+    public string GetIPAddress(String netInterfaceType, AddressFamily adressFamily = AddressFamily.InterNetwork)
+    {
+      string ipAddress = null;
+      if (Application.platform == RuntimePlatform.IPhonePlayer)
+      {
+        ipAddress = _getIPAddress(netInterfaceType);
+      }
+      return ipAddress;
+    }
+
+    public bool HasWifi()
+    {
+      bool isWifi = false;
+      if (Application.platform == RuntimePlatform.IPhonePlayer)
+      {
+        isWifi = _isWifi();
+      }
+      return isWifi;
+    }
+
+    public bool HasCellular()
+    {
+      bool isCellular = false;
+      if (Application.platform == RuntimePlatform.IPhonePlayer)
+      {
+        isCellular = _isCellular();
```

Comment:
Odd naming change here.

Reply:
OK...you might be missing some stuff from the Completed IOS code in your check-in.

Reply:
ya good catch